framework/src/wifi_manager/wifi_manager.c

```diff
@@ -489,284 +590,353 @@ wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
 
 	if (g_dw_nwevent_mqfd == (mqd_t)ERROR) {
 		ndbg("iotivity connect event message queue init fail");
-		goto error_with_mutex_destory_all;
+		return WIFI_MANAGER_FAIL;
```

Comment:
maybe call wifi_utils_deinit()?

Reply:
I moved the location of call the functions